src/orion/algo/tpe.py

```diff
@@ -630,20 +700,23 @@ def sample(self, num=1, attempts=None):
                     f"Failed to sample in interval ({self.low}, {self.high})"
                 )
             elif index is None:
+                # TODO: Why is this recursive? What is the proper return type of this function?
```

Comment:
That's to handle situations where we sample outside of the bounds. When TPE is converging to a very narrow space it becomes difficult to sample within that space.

Reply:
Fixed in 9711d1737afae8adba70986762e8219f4a268a66 (not pushed at the time of writing this)